controllers/cnfappmac_controller.go

```diff
@@ -141,80 +161,49 @@ func (r *CNFAppMacReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	var resourcesMapList []map[string]interface{}
-	podName := req.NamespacedName.Name
-	namespace := req.NamespacedName.Namespace
-	for _, nwName := range nwNameList {
-		// Get Resource name from the network name
-		netAttach := &unstructured.Unstructured{}
-		netAttach.SetKind("NetworkAttachmentDefinition")
-		netAttach.SetAPIVersion("k8s.cni.cncf.io/v1")
-		nmName := req.NamespacedName
-		nmName.Name = nwName
-		err = r.Get(ctx, nmName, netAttach)
-		if err != nil {
-			return ctrl.Result{}, err
+	if len(networks) > 0 {
+		var nwNameList []string
+		for _, item := range networks {
+			log.Info("Newtorks", "name", item["name"])
+			if !containsString(nwNameList, item["name"].(string)) {
+				nwNameList = append(nwNameList, item["name"].(string))
+			}
 		}
-		resName := netAttach.GetAnnotations()["k8s.v1.cni.cncf.io/resourceName"]
-		resName = strings.ReplaceAll(resName, "/", "_")
-		resName = strings.ReplaceAll(resName, "-", "_")
-		resName = strings.ReplaceAll(resName, ".", "_")
-		resName = strings.ToUpper(resName)
-		envName := "PCIDEVICE_" + resName
-
-		pciValue, err := getContainerEnvValue(podName, namespace, envName)
+
+		for _, nwName := range nwNameList {
+			// Get Resource name from the network name
+			netAttach := &unstructured.Unstructured{}
+			netAttach.SetKind("NetworkAttachmentDefinition")
+			netAttach.SetAPIVersion("k8s.cni.cncf.io/v1")
+			nmName := req.NamespacedName
+			nmName.Name = nwName
+			err = r.Get(ctx, nmName, netAttach)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			resName := netAttach.GetAnnotations()["k8s.v1.cni.cncf.io/resourceName"]
+			resourcesMap, _ := r.getResMap(resName, podName, namespace, nwName)
+			resourcesMapList = append(resourcesMapList, resourcesMap)
+		}
+	} else {
+		resStr, err := getContainerEnvValue(podName, namespace, "NETWORK_NAME_LIST")
+		log.Info("Resources", "NETWORK_NAME_LIST", resStr)
 		if err != nil {
+			log.Error(err, "Failed to get env NETWORK_NAME_LIST")
 			return ctrl.Result{}, err
 		}
-		pciValue = strings.TrimSuffix(pciValue, "\n")
-		pciValue = strings.TrimSuffix(pciValue, "\r")
-		pciList := strings.Split(pciValue, ",")
-		resource := nwName + "," + netAttach.GetAnnotations()["k8s.v1.cni.cncf.io/resourceName"]
-		resourcesMap := map[string]interface{}{
-			"name": nwName,
-			"res":  netAttach.GetAnnotations()["k8s.v1.cni.cncf.io/resourceName"],
-			"pci":  pciList,
-		}
-		resourcesMapList = append(resourcesMapList, resourcesMap)
-		for _, pci := range pciList {
-			resource += "," + pci
+		resList := strings.Split(strings.ReplaceAll(resStr, "\r\n", ""), ",")
+		for _, resName := range resList {
+			resourcesMap, _ := r.getResMap(resName, podName, namespace, resName)
+			resourcesMapList = append(resourcesMapList, resourcesMap)
 		}
-		resources = append(resources, resource)
 	}
 
-	macStr, err := getContainerLogValue(req.NamespacedName.Name, req.NamespacedName.Namespace)
 	//err = r.createMacFetchJob(req, pod.Spec.NodeName, resources)
```

Comment:
Since `resources` is not defined, I think this comment can be removed as well.

Reply:
I want to keep it as a reference for a while and then cleanup. Those function can also be removed.